docs/design/operator.md

```diff
@@ -0,0 +1,387 @@
+# Ceph CSI Operator Design Document
+
+## Introduction
+
+This document outlines the design of the Ceph-CSI Operator, which aims to
+provide Kubernetes native management interfaces for Ceph-CSI drivers (CephFS,
+RBD, and NFS) for Kubernetes based environments. The operator automates the
+deployment, configuration, and management of these drivers using new Kubernetes
+APIs defined as a set of Custom Resource Definitions (CRDs).
+
+## System Overview
+
+The Ceph-CSI Operator is a Kubernetes operator that offers a Kubernetes native
+management interfaces for Ceph CSI drivers. It provides a set of new Kubernetes
+APIs (CRDs) for governing deployment and management of Ceph-CSI drivers, as
+well as connection details to Ceph clusters. The operator ensures that the
+desired state specified in the CRDs is maintained in the cluster.
+
+## Assumptions and Dependencies
+
+Kubernetes cluster is available and operational. Ceph cluster is set up and
+accessible from the Kubernetes cluster. Users are familiar with Kubernetes and
+Ceph concepts. CRD and operator versions are compatible with the Kubernetes
+version in use.
+
+## Goals and Guidelines
+
+- Automate the deployment and lifecycle for Ceph-CSI drivers.
+- Provide a consistent and flexible configuration mechanism using CRDs.
+- Enable namespace-scoped configurations for different Ceph-CSI drivers.
+- Ensure high availability and scalability of the operator.
+- Support multiple deployment modes for the operator:
+  - Own-namespace: the operator manages a single Ceph-CSI driver in the current
+    namespace where operator is deployed.
+  - Multi-namespace: the operator manages Ceph-CSI drivers in user-specified
+    namespaces.
+  - All-namespaces: the operator manages Ceph-CSI drivers in all namespaces.
```

Comment:
Where is the API control for this? I can't find it in the doc.

Reply:
This doc doesnt talk about the API control for this, that going to come up in followup PR's.